### PR TITLE
Support Cn, n>9 in plot() shorthand format.

### LINKED ISSUE #27828: [Bug]: ".C10" does not work as plot shorthand format spec
### Bug summary

The plot shorthand format spec does not work when combining a Cn color with n>=10 with a linestyle or a markerstyle, e.g. ".C10".  Somewhat inconsistently, it does work when a Cn color is given by itself.

### Code for reproduction

```Python
from pylab import *; plot([0, 1], ".C10") 
# plot([0, 1], "C10") works; note that this also goes through the shorthand route
# unlike plot([0, 1], c="C10")
```

### Actual outcome

ValueError: '.C10' is not a valid format string (unrecognized character '0')

### Expected outcome

No error.

### Additional information

_No response_

### Operating system

macos

### Matplotlib Version

3.9.0.dev1178+g88d64e5f0c

### Matplotlib Backend

any

### Python version

3.12

### Jupyter version

none

### Installation

git checkout

### PR DESCRIPTION
**Pull request by anntzer**
## PR summary

Closes #27828.

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by ksunden** [comment]
It is a corner case that I'm not too sure we are worried about, but does induce a change in behavior:

`plot([2, 5, 1], "-C11")` is currently interpreted as `ls=-`, `c=C1`, `ms=1` ("tri-down" marker... which is unfortunately represented by a digit) 1,2,3,4, (tri in each direction) and 8 (octagon) each have markers associated with their string representations, so it is not _only_ tri-down (1).

With this PR it will be interpreted as a solid line of color C11, with no marker.

A workaround would be to specify the marker anywhere other than immediately after the color specification (`1-C1`, `-1C1`, `C1-1` should all work).

I will also note that `C11` goes through the fast path and does not get a marker, so there is already some inconsistency if you are using the tri-down marker which induces order dependence. (And so this only matters in the case where you also want a line and not just markers)

How concerned are we with this case?

Personally I'm of the opinion that color cycles are allowed to be longer than 10 colors, and we should support specifying them here, and thus I think I'm okay with it, but do want it acknowledged. (I'm also wondering if that change of behavior, however slight, requires a release note)

Other than the behavior change, the implementation looks good.

**Comment by timhoffm** [comment]
I’m ok with the change. The old interpretation of “C11” is not intuitive. Ideally, people will already have used a different order, at least I would have done that. But anyway it’s quite a niche case and not worth blocking the CN extension.

**Comment by tacaswell** [comment]
It needs a clear API change not, but I'm also onboard with this change.

**Comment by anntzer** [comment]
Added API change note.

File: doc/api/next_api_changes/behavior/27943-AL.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+plot() shorthand format interprets "Cn" (n>9) as a color-cycle color
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Previously, ``plot(..., "-C11")`` would be interpreted as requesting a plot
+using linestyle "-", color "C1" (1st color of the color cycle), and marker "1"
```
[the remark this answers is not in the document]
Comment (anntzer) [reply doc/api/next_api_changes/behavior/27943-AL.rst]: Switched to `color #1` which keeps the indexing consistent.

**Review by timhoffm — APPROVED** [review]

**Pull request merged**